### PR TITLE
Small docks refactor

### PR DESCRIPTION
**Pull request by malpunek**
- Cleans up View class and rename it to IconAwareView
- Moves responsibility of creating hexEdit and Visualization tabs to mainWindowWithDetachable...
- new function MainWindowWith..:: wrapWithDock() for better code reuse
- enabled Log&Database Tabs closability

### CODE REVIEW

**Comment by mkow** [comment]
Please run make format on the code (there are warnings on Jenkins about this).
You should also fix your mail in commits metadata, GitHub doesn't link them with your account ;)

File: src/ui/dockwidget.cc [hunk 1]
```diff
@@ -725,7 +704,8 @@ MainWindowWithDetachableDockWidgets::~MainWindowWithDetachableDockWidgets() {
   }
 }
 
-DockWidget* MainWindowWithDetachableDockWidgets::wrapWithDock(QWidget* widget, const QString& title){
+DockWidget* MainWindowWithDetachableDockWidgets::wrapWithDock(
+    QWidget* widget, const QString& title) {
```
Comment (mkow) [inline src/ui/dockwidget.cc]: wrapping should be 4 spaces (and 2 for indentation)

File: src/ui/dockwidget.cc [hunk 2]
```diff
@@ -1330,20 +1310,39 @@ bool MainWindowWithDetachableDockWidgets::splitDockWidgetImpl(
   return !main_window->tabifiedDockWidgets(first).contains(second);
 }
 
-void MainWindowWithDetachableDockWidgets::createHexEditTab(const QSharedPointer<FileBlobModel>& data_model) {
+void MainWindowWithDetachableDockWidgets::createHexEditTab(
+    const QSharedPointer<FileBlobModel>& data_model) {
```
Comment (mkow) [inline src/ui/dockwidget.cc]: ditto

File: src/ui/dockwidget.cc [hunk 3]
```diff
@@ -1330,20 +1310,39 @@ bool MainWindowWithDetachableDockWidgets::splitDockWidgetImpl(
...
       new QItemSelectionModel(data_model.data()));
 
   auto* node_widget = new NodeWidget(this, data_model, selection_model);
   addTab(node_widget, data_model->path().join(" : "), nullptr);
 }
 
-void MainWindowWithDetachableDockWidgets::createHexEditTab(const QString& fileName,
-                                       const dbif::ObjectHandle& fileBlob) {
+void MainWindowWithDetachableDockWidgets::createHexEditTab(
+    const QString& fileName, const dbif::ObjectHandle& fileBlob) {
```
Comment (mkow) [inline src/ui/dockwidget.cc]: ditto

File: src/ui/dockwidget.cc (hunk 1 above)
Comment (mkow) [reply src/ui/dockwidget.cc]: Please disregard this comment, the format is ok here.

File: src/ui/dockwidget.cc (hunk 2 above)
Comment (mkow) [reply src/ui/dockwidget.cc]: Please disregard this comment, the format is ok here.

File: src/ui/dockwidget.cc [hunk 4]
```diff
@@ -1335,5 +1330,20 @@ bool MainWindowWithDetachableDockWidgets::splitDockWidgetImpl(
...
+void MainWindowWithDetachableDockWidgets::createHexEditTab(const QSharedPointer<FileBlobModel>& data_model) {
+  QSharedPointer<QItemSelectionModel> selection_model(
+      new QItemSelectionModel(data_model.data()));
+
+  auto* node_widget = new NodeWidget(this, data_model, selection_model);
+  addTab(node_widget, data_model->path().join(" : "), nullptr);
+}
+
+void MainWindowWithDetachableDockWidgets::createHexEditTab(const QString& fileName,
+                                       const dbif::ObjectHandle& fileBlob) {
```
[the remark this answers is not in the document]
Comment (mkow) [reply src/ui/dockwidget.cc]: Please disregard this comment, the format is ok here.

File: src/ui/dockwidget.cc (hunk 3 above)
Comment (mkow) [reply src/ui/dockwidget.cc]: Please disregard this comment, the format is ok here.

File: src/ui/veles_mainwindow.cc [hunk 5]
```diff
@@ -404,24 +399,23 @@ void VelesMainWindow::createFileBlob(const QString& file_name) {
...
   QSharedPointer<QItemSelectionModel> selection_model(
       new QItemSelectionModel(data_model.data()));
 
   auto* node_widget = new NodeWidget(this, data_model, selection_model);
   addTab(node_widget, data_model->path().join(" : "), nullptr);
 }
 
+void VelesMainWindow::createHexEditTab(const QString& fileName,
+                                       const dbif::ObjectHandle& fileBlob) {
+  QSharedPointer<FileBlobModel> data_model(
```
[the remark this answers is not in the document]
Comment (malpunek) [reply src/ui/veles_mainwindow.cc]: FIXME became irrelevant. The line suggested `// auto sibling = DockWidget::getParentDockWidget(this);` doesn't even make sense. Sibiling == Parent ???

File: include/ui/dockwidget.h [hunk 6]
```diff
@@ -147,15 +147,15 @@ class TabBarEventFilter : public QObject {
 };
 
 /*****************************************************************************/
-/* View */
+/* IconAwareView */
```
[the remark this answers is not in the document]
Comment (malpunek) [reply include/ui/dockwidget.h]: done

File: include/ui/mainwindowwithdetachabledockwidgets.h [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+  TabBarEventFilter* tab_bar_event_filter_;
+  QRubberBand* rubber_band_;
+
+  bool dock_widgets_with_no_title_bars_ = false;
+  bool icons_on_tabs_;
+  bool mark_active_dock_widget_;
+};
+}
+}
```
Comment (mkow) [inline include/ui/mainwindowwithdetachabledockwidgets.h]: Please add a newline to the end of this file.

File: include/ui/mainwindowwithdetachabledockwidgets.h [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+  static int last_created_window_id_;
+  static QPointer<DockWidget> active_dock_widget_;
+
+  TabBarEventFilter* tab_bar_event_filter_;
+  QRubberBand* rubber_band_;
+
+  bool dock_widgets_with_no_title_bars_ = false;
+  bool icons_on_tabs_;
+  bool mark_active_dock_widget_;
+};
```
Comment (mkow) [inline include/ui/mainwindowwithdetachabledockwidgets.h]: Add a newline between class end and namespaces ends

File: include/ui/mainwindowwithdetachabledockwidgets.h (hunk 7 above)
Comment (mkow) [reply include/ui/mainwindowwithdetachabledockwidgets.h]: Namespaces should end with a proper comment (https://google.github.io/styleguide/cppguide.html#Namespaces)

File: src/ui/hexeditwidget.cc [hunk 9]
```diff
@@ -361,11 +361,11 @@ void HexEditWidget::saveAs() {
 }
 
 void HexEditWidget::showVisualization() {
```
Comment (mkow) [inline src/ui/hexeditwidget.cc]: Please also fix this name (but in a separate PR)

File: src/ui/mainwindowwithdetachabledockwidgets.cc [hunk 10]
```diff
@@ -0,0 +1,717 @@
...
+  // FIXME: main_window_ needs to be updated when docks are moved around,
+  // then we can use this behaviour without any weird effects
+  // auto sibling = DockWidget::getParentDockWidget(this);
+
+  auto dock_widget = addTab(panel, data_model->path().join(" : "), nullptr);
+  connect(dock_widget, &DockWidget::visibilityChanged, panel,
+          &visualization::VisualizationPanel::visibilityChanged);
+}
+}
+}
```
Comment (mkow) [inline src/ui/mainwindowwithdetachabledockwidgets.cc]: Newline

File: src/ui/mainwindowwithdetachabledockwidgets.cc [hunk 11]
```diff
@@ -0,0 +1,717 @@
...
+  panel->setAttribute(Qt::WA_DeleteOnClose);
+
+  // FIXME: main_window_ needs to be updated when docks are moved around,
+  // then we can use this behaviour without any weird effects
+  // auto sibling = DockWidget::getParentDockWidget(this);
+
+  auto dock_widget = addTab(panel, data_model->path().join(" : "), nullptr);
+  connect(dock_widget, &DockWidget::visibilityChanged, panel,
+          &visualization::VisualizationPanel::visibilityChanged);
+}
```
Comment (mkow) [inline src/ui/mainwindowwithdetachabledockwidgets.cc]: Newline

File: src/ui/mainwindowwithdetachabledockwidgets.cc (hunk 10 above)
Comment (mkow) [reply src/ui/mainwindowwithdetachabledockwidgets.cc]: namespace comments

File: src/ui/mainwindowwithdetachabledockwidgets.cc [hunk 12]
```diff
@@ -0,0 +1,717 @@
...
+  auto* panel = new visualization::VisualizationPanel(this, data_model);
+
+  panel->setData(
+      QByteArray(reinterpret_cast<const char*>(data_model->binData().rawData()),
+                 static_cast<int>(data_model->binData().size())));
+  panel->setAttribute(Qt::WA_DeleteOnClose);
+
+  // FIXME: main_window_ needs to be updated when docks are moved around,
+  // then we can use this behaviour without any weird effects
+  // auto sibling = DockWidget::getParentDockWidget(this);
```
Comment (mkow) [inline src/ui/mainwindowwithdetachabledockwidgets.cc]: Hmm, you've removed this FIXME from another place, if it doesn't make sense anymore please also remove it from here ;)

Comment (malpunek) [reply src/ui/mainwindowwithdetachabledockwidgets.cc]: Actually it might make sense here. I hope to actually fix it in next PR

Comment (malpunek) [reply src/ui/mainwindowwithdetachabledockwidgets.cc]: Well ok, actually it KIND OF make sense. The commented code is invalid, yet I would like to leave a mention about actually passing a pointer to sibling here. I love focus on this in one of my next PRs

Comment (mkow) [reply src/ui/mainwindowwithdetachabledockwidgets.cc]: Sure, let's deal with it in another PR.

**Pull request merged**